ml-agents/mlagents/trainers/trainer_controller.py

```diff
@@ -49,6 +50,8 @@ def __init__(
         :param resampling_interval: Specifies number of simulation steps after which reset parameters are resampled.
         """
         self.trainers: Dict[str, Trainer] = {}
+        self.multi_trainers: Dict[str, Trainer] = {}
```

Comment:
Do some trainers go in `trainers` and some in `multi_trainers`?

Reply:
This actually doesn't get used.  All trainers go into `self.trainers` corresponding to root behavior name.  Then, I use a separate `dict[str, set]` which maps root behavior name to `name behavior id`. Removing this.